mono/metadata/marshal.h

```diff
@@ -247,13 +247,78 @@ typedef struct {
 	} d;
 } WrapperInfo;
 
+enum {
+	STELEMREF_OBJECT, /*no check at all*/
+	STELEMREF_SEALED_CLASS, /*check vtable->klass->element_type */
+	STELEMREF_CLASS, /*only the klass->parents check*/
+	STELEMREF_CLASS_SMALL_IDEPTH, /* like STELEMREF_CLASS bit without the idepth check */
+	STELEMREF_INTERFACE, /*interfaces without variant generic arguments. */
+	STELEMREF_COMPLEX, /*arrays, MBR or types with variant generic args - go straight to icalls*/
+	STELEMREF_KIND_COUNT
+};
+
+static const char *strelemref_wrapper_name[] = {
```

Comment:
Do we need this random constant on the header?
We should wrap it under a `mono_marshal_get_strelemref_wrapper_name`

Reply:
good point, let me do that in a follow up commit in master.